### ISSUE TITLE
3199 (and before) - Map Creator problems

### ISSUE THREAD

**Issue by panther2** [triplea-game/triplea#1261]
- map.properties process does not show all options, an empty file is created
- Auto PlacementFinder does not find the map when using the new folder structure. Using the old folder structure it freezes (**solved (April 03 2017)** see starting from https://github.com/triplea-game/triplea/issues/1261#issuecomment-290971708 ).

I stopped trying it out at that point.

What has been working in 1.8.0.9 fails in 1.9.x

Issue #1190 is relevant in this context, too.
Users creating maps with 1.8.0.9 need to know how to "convert" them to 1.9 then.

**Comment by DanVanAtta** [comment]
This is using the 'old' map creator?

**Comment by panther2** [comment]
Yes -> Engine Preferences -> Run the Map Creator

I have now idea how the beta one works... I just realized that it starts with selecting a game.xml (which at the time of creating a mpa from scratch is not yet present). Can you point me to a documentation?

**Comment by Cernelius** [comment]
Yeah, before releasing somebody reliable should try fully making up a map with the candidate. Didn't Frostion use the prerelease when he made his new Iron War map?

**Comment by DanVanAtta** [comment]
Map creator has had issues, probably even more so now. Fixes to it can be done as patches without impacting save game compatibility. This does not affect actual game play, so not a blocker for having players migrate to 1.9

A bug report listing the items that are not working would be welcome. Requesting it contain relatively detailed instructions on how to reproduce the problems. The detailed steps are to simplify reproduction, and to also ensure that we are actually doing the same process/thing. IE: my definition of "map.properties process does not show all options", might be different, and that would lead to confusion. This is probably a good issue to list the  problems with map creator.

**Comment by panther2** [comment]
I will further elaborate on this issue, as soon as I find some more time to do so.
I agree it is no 1.9 blocker, as we can use 1.8.0.9 for map creating. It would be important to know about the necessary steps to convert 1.8-maps to 1.9-maps later.

**Comment by FrostionAAA** [comment]
@Cernelius and all, no, I did not use any map creator to make the XML for Iron War. I think I tried it out for like 5 minuttes, ran into problems and decided to just modify another old XML to fit the new map.

I did of course use all the other old tools (they are essential).

**Comment by ssoloff** [comment]
@panther2, let me ask you a question about your second problem:

> `AutoPlacementFinder` does not find the map when using the new folder structure. Using the old folder structure it freezes.

Can you confirm that other tools (e.g. `CenterPicker`, `DecorationPicker`, etc.) work for you with either the old or new maps structure?

I'm asking because I think I was a bit confused earlier today in our previous [discussion](https://github.com/triplea-game/triplea/pull/1602#issuecomment-290930893).  When I stated all of the other tools had a similar problem to what you described above, it was because I didn't realize the relationship between the old and new map structures: namely, that the `<mapDir>/map` folder in the new structure is basically the content of the old structure.  For example, consider `great_war`.  In the old structure, `map.properties` would have been located at `<mapsDir>/great_war/map.properties`.  While in the new structure, the same file is located at `<mapsDir>/great_war/map/map.properties`.  Is that correct?

If true, then I misspoke before.  My confusion was due to the fact that I was using `great_war` for testing, and that map doesn't have a "map image" (e.g. `baseTiles.png` or something) in the `map` folder.  So I just assumed (incorrectly) that these tools were prompting me for the root directory of the entire map download, not the `map` directory under that root. 👎 

Anyway, that means that `AutoPlacementFinder` is probably the only tool broken at this point by the new map structure.  Specifically because it uses a totally different mechanism from the other tools for determining where the map files are located.  Basically, it doesn't know about the new `map` folder.  The other tools don't explicitly know about it either, but they allow the user to navigate to the correct folder via a file chooser dialog.

Modifying `AutoPlacementFinder` to handle either the old or new map structure is relatively straightforward.  However, please let me know if you believe there's something I'm overlooking.

**Comment by panther2** [comment]
@ssoloff 

I retested with 3946 again - using `minimap` as there is a complete border file present in the map directory.

I can confirm, that **almost** everything - including `AutoPlacementFinder` is working fine for me now - it seems that the new folder structure is recognized correctly now.

The only aspect that is still not working is the `map.properties` process. As indicated in the first post, only partial options are offered and an empty file is saved.

I did not test the old folder structure though - is this still used? 

Hope that helps :smiley:

**Comment by panther2** [comment]
For further illustration:

This is the Map Properties Maker in 1.8.0.9:

![1809](https://cloud.githubusercontent.com/assets/20812775/24589354/39741446-17d9-11e7-8e69-9f26a2c8b8e9.gif)

and this is the Map Properties Maker in 1.9.0.0.3946:

![1900](https://cloud.githubusercontent.com/assets/20812775/24589361/5325ccf4-17d9-11e7-87f5-6b4534be0aaa.gif)

**Comment by panther2** [comment]
@ssoloff 

> When using `AutoPlacementFinder`, can you confirm you see the following dialog immediately after selecting `minimap` as the map to edit.

So this is how it goes here:
![1900a](https://cloud.githubusercontent.com/assets/20812775/24596308/b10237c6-183e-11e7-9e5e-f8d680207b4b.gif)

Looks fine using the new folder structure...

Thank you for taking care.

**Comment by panther2** [comment]
Accidentally hit `close`.

**Comment by panther2** [comment]
@ssoloff 

Please note that I have edited the above posting to add some information.

This is how it looks like when I skip the "Step 1 Map Properties"
and got to "Step 2 Map Utilities" without saving the map place before:

![1900b](https://cloud.githubusercontent.com/assets/20812775/24596530/31483448-1840-11e7-9673-6ddaed6536a1.gif)

**Comment by ssoloff** [comment]
@panther2 

> Looks fine using the new folder structure...

Ahh...  You're starting the `AutoPlacementFinder` tool from Map Creator.  If I actually tried to create a map once in my life, I would have assumed that. 😃 I've been running `AutoPlacementFinder` from the command line with no additional arguments.  I think that explains why we're seeing different behavior.

I believe I understand why it's working for you based on both of your screencasts.  Even if the map folder is not specified, Map Creator **is** specifying the `triplea.unit.*` properties when it starts `AutoPlacementFinder`.  When any of those are present, you do not get prompted to specify those properties via the UI **even if `map.properties` cannot be located**.

However, when running `AutoPlacementFinder` from the command line with no arguments, and then choosing a map that has the new folder structure when prompted, it cannot locate `map.properties` and thus prompts you to input the unit properties.  I went ahead and submitted #1607 to fix this issue, even though my use case is probably unlikely to ever occur in practice.
